### DISCUSSION TITLE
Module transpiling makes it impossible to use most libraries in custom endpoints

### DISCUSSION THREAD

**Discussion by goldenpath-ai** [directus/directus#16408] [Archive]
Many, if not most node libraries won't work at all in custom endpoints. Sometimes the module builds but then doesn't work and sometimes it doesn't build at all. Making custom endpoints pretty much unusable in a lot of cases.

1. Transpiling doesn't fix relative imports - if a library references its own files that are non-js the imports don't work. This also doesn't produce any errors or warnings, I had to figure that out for myself
    - library: `pdf-parse`
    
2. Sometimes it doesn't transpile at all - errors lUnexpected character '�' (Note that you need plugins to import files that are not JavaScript) (plugin rpt2)
   - library `ssh2-sftp-client`

### To Reproduce

Create a custom endpoint, install and import any of the libraries mentioned above

### Errors Shown

In case n. 2:

`[RollupError] Unexpected character '�' (Note that you need plugins to import files that are not JavaScript) (plugin rpt2)`

**Comment by paescuj** [comment]
Not sure if you already are aware of this, but you can configure additional plugins like this: https://docs.directus.io/extensions/creating-extensions.html#configuring-the-cli
